asdf/commands/tests/test_edit.py

```diff
@@ -0,0 +1,97 @@
+import os
+import re
+
+import numpy as np
+import pytest
+
+import asdf
+from asdf.commands import main
+
+
+def _create_base_asdf(version, oname):
+    """
+    In the test temp directory, create a base ASDF file to edit
+    and test against.
+    """
+    seq = np.arange(100)
+
+    # Store the data in an arbitrarily nested dictionary
+    tree = {
+        "foo": 42,
+        "name": "Monty",
+        "sequence": seq,
+    }
+
+    with asdf.AsdfFile(tree, version=version) as af:
+        af.write_to(oname)
+
+
+def _create_edited_yaml(base_yaml, edited_yaml, pattern, replacement):
+    with open(base_yaml, "rb") as fd:
+        content = fd.read()
+        new_content = re.sub(pattern, replacement, content)
+        with open(edited_yaml, "wb") as fd:
+            fd.write(new_content)
+
+
+def _initialize_test(tmpdir, version, test_name):
```

Comment:
`test_name` is not needed here, as `tmpdir` always creates a new temporary directory under `basetemp` for each test (and each parameterization), so files will be segregated by directory based on the test name and parameter (version in this case).  Changing the filename is redundant.

Reply:
I added this due to the Windows problems I was having during Travis-CI testing.  I removed it.